packages/server/hubspot/hubSpotApi.ts

```diff
@@ -0,0 +1,143 @@
+import fetch from 'node-fetch'
+const accessToken = process.env.HUBSPOT_API_KEY
+
+type Company = {
+  id: string
+  archived: boolean
+}
+
+type DealsResponse = {
+  total: number
+}
+
+type CompaniesResponse = {
+  results: Company[]
+}
+
+const getCompaniesByDomain = async (domain: string): Promise<CompaniesResponse | Error> => {
+  const body = JSON.stringify({
+    properties: ['domain', 'hs_additional_domains'],
+    filterGroups: [
+      {
+        filters: [
+          {
+            operator: 'EQ',
+            propertyName: 'domain',
+            value: domain
+          }
+        ]
+      },
+      {
+        filters: [
+          {
+            operator: 'CONTAINS_TOKEN',
+            propertyName: 'hs_additional_domains',
+            value: domain
+          }
+        ]
+      }
+    ],
+    limit: 100
+  })
+
+  const res = await fetch(`https://api.hubapi.com/crm/v3/objects/companies/search`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${accessToken}`
+    },
+    body
+  })
+
+  const resJson = await res.json()
+
+  if (resJson.status === 'error') {
+    return new Error(resJson.message)
+  }
+
+  return resJson
+}
+
+const SALES_PIPELINE_ID = '14fa7db3-cc82-4562-8603-90d792014b42'
```

Comment:
+1 I think these ids should probably be stored in the `.env` file. They're useless without the `HUBSPOT_API_KEY`, but if someone got access to that, they could quickly pull all of our sales data

Reply:
-1 Yes, let's prefix this with HUBSPOT and put in the `.env`